pkg/kor/kor.go

```diff
@@ -56,17 +61,31 @@ func GetKubeClient(kubeconfig string) *kubernetes.Clientset {
 	return clientset
 }
 
-func SetNamespaceList(namespace string, kubeClient *kubernetes.Clientset) []string {
-	var namespaces []string
-	if namespace != "" {
-		namespaces = append(namespaces, namespace)
-	} else {
-		namespaceList, err := kubeClient.CoreV1().Namespaces().List(context.TODO(), metav1.ListOptions{})
-		if err != nil {
-			fmt.Fprintf(os.Stderr, "Failed to retrieve namespaces: %v\n", err)
-			os.Exit(1)
+func SetNamespaceList(namespaceLists IncludeExcludeLists, kubeClient *kubernetes.Clientset) []string {
+	namespaces := make([]string, 0)
+
+	includeNamespaceMap := make(map[string]struct{})
+	excludeNamespaceMap := make(map[string]struct{})
+	includeNamespaces := strings.Split(namespaceLists.IncludeListStr, ",")
+	excludeNamespaces := strings.Split(namespaceLists.ExcludeListStr, ",")
+	for _, ns := range excludeNamespaces {
+		excludeNamespaceMap[ns] = struct{}{}
+	}
+	for _, ns := range includeNamespaces {
+		includeNamespaceMap[ns] = struct{}{}
+	}
+	namespaceList, err := kubeClient.CoreV1().Namespaces().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Failed to retrieve namespaces: %v\n", err)
+		os.Exit(1)
+	}
+	for _, ns := range namespaceList.Items {
+		if len(includeNamespaces) == 0 {
```

Comment:
`len(includeNamespaces)` is never equal to 0.
If it is empty it will just contain ""
you can try `includeNamespaces[0] != ""`